File: web/src/api/skill.ts
import request, { type ListQuery } from './request'

export interface Skill {
  id: number
  uuid: string
  name: string
  description: string
  instruction: string
  tools?: any[]
  created_at: string
  updated_at: string
}

export interface CreateSkillReq {
  name: string
  description?: string
  instruction?: string
  tool_ids?: number[]
}

export const skillApi = {
  list: (params: ListQuery) => request.get('/skills', { params }),
  get: (id: number) => request.get(`/skills/${id}`),
  create: (data: CreateSkillReq) => request.post('/skills', data),
  update: (id: number, data: Partial<CreateSkillReq>) => request.put(`/skills/${id}`, data),
  delete: (id: number) => request.delete(`/skills/${id}`),
}

File: internal/model/skill.go
package model

import "time"

type Skill struct {
	ID          int64     `json:"id"`
	UUID        string    `json:"uuid"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Instruction string    `json:"instruction"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	Tools []Tool `json:"tools,omitzero"`
}

type CreateSkillReq struct {
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Instruction string  `json:"instruction"`
	ToolIDs     []int64 `json:"tool_ids,omitzero"`
}

type UpdateSkillReq struct {
	Name        *string `json:"name,omitzero"`
	Description *string `json:"description,omitzero"`
	Instruction *string `json:"instruction,omitzero"`
	ToolIDs     []int64 `json:"tool_ids,omitzero"`
}
